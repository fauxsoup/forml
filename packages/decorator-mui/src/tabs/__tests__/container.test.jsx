import { Container, Tab, Panel } from '../';
import Context from '@forml/context';
import React from 'react';
import { render } from '@testing-library/react';
import * as decorator from '../';

describe('renders', function () {
    let form;
    let title = 'title';
    let description = 'description';
    let tabs = [];
    let panels = [];

    beforeEach(function () {
        form = {
            type: 'container',
            tabs: [{ type: 'fieldset', items: [{ key: [] }] }],
        };
        tabs = [
            <Tab form={form.tabs[0]} parent={form}>
                Test Tab
            </Tab>,
        ];
        panels = [
            <Panel form={form.tabs[0]} parent={form}>
                Test Panel
            </Panel>,
        ];
    });

    describe('with form options', function () {
        let fields = {
            layout: ['horizontal', 'vertical'],
        };

        Object.keys(fields).forEach(function (field) {
            fields[field].forEach(function (value) {
                describe(`${field}`, function () {
                    test(`${value}`, function () {
                        form = { ...form, [field]: value };
                        const { container } = render(
                            <Context.Provider value={{ decorator }}>
                                <Container
                                    form={form}
                                    title={title}
                                    description={description}
                                    tabs={tabs}
                                    panels={panels}
                                />
                            </Context.Provider>
                        );

                        expect(container).toMatchSnapshot();
                    });
                });
            });
        });
    });

    test('with title and description', function () {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <Container
                    form={form}
                    title={title}
                    description={description}
                    tabs={tabs}
                    panels={panels}
                />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test('with title and no description', function () {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <Container
                    form={form}
                    title={title}
                    tabs={tabs}
                    panels={panels}
                />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test('with description and no title', function () {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <Container
                    form={form}
                    description={title}
                    tabs={tabs}
                    panels={panels}
                />
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });

    test('with no title or description', function () {
        const { container } = render(
            <Context.Provider value={{ decorator }}>
                <Container form={form} tabs={tabs} panels={panels}></Container>
            </Context.Provider>
        );

        expect(container).toMatchSnapshot();
    });
});
