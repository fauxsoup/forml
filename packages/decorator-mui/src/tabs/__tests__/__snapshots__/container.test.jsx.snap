// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with description and no title 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-77 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-titleRoot-82 MuiList-dense"
    >
      <li
        class="MuiListItem-root makeStyles-title-85 MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root MuiIcon-fontSizeSmall"
          >
            view_carousel
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <span
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-displayBlock"
          >
            title
          </span>
        </div>
      </li>
    </ul>
    <div
      class="makeStyles-content-78 makeStyles-vertical-84"
    >
      <div
        class="MuiPaper-root makeStyles-tabs-80 makeStyles-vertical-84 MuiPaper-elevation0"
      >
        <ul
          class="MuiList-root MuiList-dense"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-root-86 makeStyles-vertical-88 MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                dynamic_form
              </span>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
      <div
        class="makeStyles-panels-79"
      >
        <div
          class="MuiPaper-root makeStyles-root-92 makeStyles-horizontal-94 MuiPaper-elevation0 MuiPaper-rounded"
          form="[object Object]"
          parent="[object Object]"
        >
          Test Panel
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders with form options layout horizontal 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-titleRoot-6 MuiList-dense"
    >
      <li
        class="MuiListItem-root makeStyles-title-9 MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root MuiIcon-fontSizeSmall"
          >
            view_carousel
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle2 MuiTypography-colorTextPrimary MuiTypography-noWrap MuiTypography-displayBlock"
          >
            title
          </span>
          <span
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-displayBlock"
          >
            description
          </span>
        </div>
      </li>
    </ul>
    <div
      class="makeStyles-content-2 makeStyles-horizontal-7"
    >
      <div
        class="MuiPaper-root makeStyles-tabs-4 makeStyles-horizontal-7 MuiPaper-elevation0"
      >
        <ul
          class="MuiList-root MuiList-dense"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-root-10 makeStyles-vertical-12 MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                dynamic_form
              </span>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
      <div
        class="makeStyles-panels-3"
      >
        <div
          class="MuiPaper-root makeStyles-root-16 makeStyles-horizontal-18 MuiPaper-elevation0 MuiPaper-rounded"
          form="[object Object]"
          parent="[object Object]"
        >
          Test Panel
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders with form options layout vertical 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-20 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-titleRoot-25 MuiList-dense"
    >
      <li
        class="MuiListItem-root makeStyles-title-28 MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root MuiIcon-fontSizeSmall"
          >
            view_carousel
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle2 MuiTypography-colorTextPrimary MuiTypography-noWrap MuiTypography-displayBlock"
          >
            title
          </span>
          <span
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-displayBlock"
          >
            description
          </span>
        </div>
      </li>
    </ul>
    <div
      class="makeStyles-content-21 makeStyles-vertical-27"
    >
      <div
        class="MuiPaper-root makeStyles-tabs-23 makeStyles-vertical-27 MuiPaper-elevation0"
      >
        <ul
          class="MuiList-root MuiList-dense"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-root-29 makeStyles-vertical-31 MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                dynamic_form
              </span>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
      <div
        class="makeStyles-panels-22"
      >
        <div
          class="MuiPaper-root makeStyles-root-35 makeStyles-horizontal-37 MuiPaper-elevation0 MuiPaper-rounded"
          form="[object Object]"
          parent="[object Object]"
        >
          Test Panel
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders with no title or description 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-96 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="makeStyles-content-97 makeStyles-vertical-103"
    >
      <div
        class="MuiPaper-root makeStyles-tabs-99 makeStyles-vertical-103 MuiPaper-elevation0"
      >
        <ul
          class="MuiList-root MuiList-dense"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-root-105 makeStyles-vertical-107 MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                dynamic_form
              </span>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
      <div
        class="makeStyles-panels-98"
      >
        <div
          class="MuiPaper-root makeStyles-root-111 makeStyles-horizontal-113 MuiPaper-elevation0 MuiPaper-rounded"
          form="[object Object]"
          parent="[object Object]"
        >
          Test Panel
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders with title and description 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-39 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-titleRoot-44 MuiList-dense"
    >
      <li
        class="MuiListItem-root makeStyles-title-47 MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root MuiIcon-fontSizeSmall"
          >
            view_carousel
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense MuiListItemText-multiline"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle2 MuiTypography-colorTextPrimary MuiTypography-noWrap MuiTypography-displayBlock"
          >
            title
          </span>
          <span
            class="MuiTypography-root MuiListItemText-secondary MuiTypography-caption MuiTypography-colorTextSecondary MuiTypography-noWrap MuiTypography-displayBlock"
          >
            description
          </span>
        </div>
      </li>
    </ul>
    <div
      class="makeStyles-content-40 makeStyles-vertical-46"
    >
      <div
        class="MuiPaper-root makeStyles-tabs-42 makeStyles-vertical-46 MuiPaper-elevation0"
      >
        <ul
          class="MuiList-root MuiList-dense"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-root-48 makeStyles-vertical-50 MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                dynamic_form
              </span>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
      <div
        class="makeStyles-panels-41"
      >
        <div
          class="MuiPaper-root makeStyles-root-54 makeStyles-horizontal-56 MuiPaper-elevation0 MuiPaper-rounded"
          form="[object Object]"
          parent="[object Object]"
        >
          Test Panel
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`renders with title and no description 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-root-58 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-titleRoot-63 MuiList-dense"
    >
      <li
        class="MuiListItem-root makeStyles-title-66 MuiListItem-dense MuiListItem-gutters MuiListItem-divider"
      >
        <div
          class="MuiListItemIcon-root"
        >
          <span
            aria-hidden="true"
            class="material-icons MuiIcon-root MuiIcon-fontSizeSmall"
          >
            view_carousel
          </span>
        </div>
        <div
          class="MuiListItemText-root MuiListItemText-dense"
        >
          <span
            class="MuiTypography-root MuiListItemText-primary MuiTypography-subtitle2 MuiTypography-colorTextPrimary MuiTypography-noWrap MuiTypography-displayBlock"
          >
            title
          </span>
        </div>
      </li>
    </ul>
    <div
      class="makeStyles-content-59 makeStyles-vertical-65"
    >
      <div
        class="MuiPaper-root makeStyles-tabs-61 makeStyles-vertical-65 MuiPaper-elevation0"
      >
        <ul
          class="MuiList-root MuiList-dense"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root makeStyles-root-67 makeStyles-vertical-69 MuiListItem-dense MuiListItem-gutters MuiListItem-divider MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemIcon-root"
            >
              <span
                aria-hidden="true"
                class="material-icons MuiIcon-root"
              >
                dynamic_form
              </span>
            </div>
            <div
              class="MuiListItemText-root MuiListItemText-dense"
            />
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </ul>
      </div>
      <div
        class="makeStyles-panels-60"
      >
        <div
          class="MuiPaper-root makeStyles-root-73 makeStyles-horizontal-75 MuiPaper-elevation0 MuiPaper-rounded"
          form="[object Object]"
          parent="[object Object]"
        >
          Test Panel
        </div>
      </div>
    </div>
  </div>
</div>
`;
